tests/unit/test_advanced.py

```diff
@@ -537,3 +537,23 @@ def test_do_attempt_keeps_highest_score(self):
             FeedbackMessages.MessageClasses.PARTIAL_SOLUTION
         )
         self.assertIn(expected_feedback, res[self.OVERALL_FEEDBACK_KEY])
+
+    def test_do_attempt_check_score_with_decoy(self):
+        self.assertFalse(self.block.completed)  # precondition check
+        expected_score = 4.0 / 5.0
+        self._submit_solution({
+            0: self.ZONE_1,
+            1: self.ZONE_2,
+            2: self.ZONE_2,
+            3: self.ZONE_1})  # incorrect solution, 0.8 score
```

Comment:
Nit: according to the [edX Python Style Guide](http://edx.readthedocs.io/projects/edx-developer-guide/en/latest/style_guides/python_guidelines.html#breaking-long-lines), the closing parens should be on a new line, and the last item in the list should preferably end with a trailing comma:

```
...
    3: self.ZONE_1,
})